enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,973 @@
+      must be created in the same namespace as the operator.
+    * When interacting with Kubernetes API, carefully choose the right kubeconfig to watch / create / update objects in
+      the right cluster.
+    * When running aws-ebs-csi-driver-operator Deployment:
+      * Give it kubeconfig to the guest cluster as a Secret volume + cmdline parameter. 
+      * Pass it Secret with STS creds generated by control-plane-operator.
+      * Add `token-minter` sidecar, which will create STS secrets to the driver.
+      * Set `priorityClassName: hypershift-control-plane`.
+    * Use `priorityClassName: hypershift-control-plane` for the driver operator.
+    * Phase 2: add metrics services + TLS keys.
+
+* **aws-ebs-csi-driver-operator**:
+  * Add a new cmdline option for the guest cluster kubeconfig file location.
+    * Parse both kubeconfigs:
+      * One from projected service account, which leads to the management cluster.
+      * Second from the new cmdline option introduced above. This one leads to the guest cluster.
+    * Rework CSI driver Deployment to run only the control plane parts of the CSI driver, i.e.:
+      * Remove `hostNetwork: true`.
+      * Add `controller` as the first command line arg.
+      * Set env. var. `AWS_REGION=us-east2` with the actual guest cluster region. The region can be read from
+        `Infrastructure` CR.
+      * This removes dependency on AWS metadata service and allows us to run the controller parts of the driver outside
+        of AWS.
+    * Rework CSI driver DaemonSet to run only node parts of the CSI driver.
+      * Add `node` as the first command line arg.
+      * This allows us not to give the DaemonSet any AWS credentials.
+  * Only on HyperShift:
+    * When interacting with Kubernetes API, carefully choose the right kubeconfig to watch / create / update objects in
+      the right cluster.
+    * Replace namespaces in the driver Deployment, PDBs and other objects that are created in the management cluster.
+      They must be created in the same namespace as the operator.
+    * Update the CSI driver control-plane Deployment: 
+      * Give it kubeconfig to the guest cluster as a Secret volume.
+      * Pass it Secret with STS creds generated by control-plane-operator.
+      * Add `token-minter` sidecar, which will create STS secrets to the driver.
+      * Set `priorityClassName: hypershift-control-plane`.
+    * Phase 2: add metrics services + TLS keys.
+
+* **AWS EBS CSI driver**:
+  * No changes there!
+    * Controller parts will run in the management cluster, but they get kubeconfig to the guest cluster.
+    * Node DaemonSet will get the default projected kubeconfig that can access objects only in the
+      openshift-cluster-csi-drivers namespace. 
+
+#### Notes about other CSI drivers
+
+* Azure Disk CSI driver
+  * It needs some cloud credentials available on nodes. We don't have exact list of permissions needed, they could be
```

Comment:
we already place credentials for azure inside the guest cluster

Reply:
Yeah, we need to review what permissions the CSI driver on the node actually needs, I am pretty sure it's not much. We don't want a compromised node to be able to delete volumes in Azure. I rephrased it a bit.